inifix/format.py

```diff
@@ -2,44 +2,86 @@
 import os
 import re
 import sys
+import warnings
 from typing import Optional
 from typing import Sequence
 
 from inifix.io import load
 
 NAME_COL_MINSIZE = 10
 COMMENT_COL = 80
+DEFAULT_PADDING_SIZE = 4
+
+
+def ljust(string: str, width: int) -> str:
+    """Similar to str.ljust, but will always append a whitespace ' ' at the end of the string, for the ini file to be usable"""
+    justified_name = string.ljust(width)
+    if (
+        len(string) < width
+    ):  # in this case, ljust already appended a whitespace : no need to append one
+        readable_name = justified_name
+    else:
+        readable_name = justified_name + " "
+    return readable_name
 
 
 def _normalize_whitespace(s: str) -> str:
     return re.sub(r"\s", " ", s).strip()
 
 
-def iniformat(data: str) -> str:
+def iniformat(data: str, *, column_length: Optional[int] = None) -> str:
     lines = data.splitlines()
     contents = []
     comments = []
+    parameters = []
+    values = []
     for line in lines:
         content, _, comment = line.partition("#")
         contents.append(_normalize_whitespace(content))
         comments.append(_normalize_whitespace(comment))
+        if not content.startswith("[") and content != "":
+            parameter, *value = content.split()
+            parameters.append(parameter)
+            values.append(value)
+
+    max_name_size = max(len(parameter) for parameter in parameters)
+
+    if column_length is None:
+        padded_name_col_size = max(
+            max_name_size + DEFAULT_PADDING_SIZE, NAME_COL_MINSIZE
+        )
+    else:
+        padded_name_col_size = column_length
+
+        if max_name_size >= column_length:
+            long_parameters = [
+                parameter for parameter in parameters if len(parameter) >= column_length
+            ]
+            warnings.warn(
+                "parameter/s:\n"
+                + "\n".join([f"{long_parameter}" for long_parameter in long_parameters])
+                + "\nwith length/s "
+                + ", ".join(
+                    [f"{len(long_parameter)}" for long_parameter in long_parameters]
+                )
+                + f" is/are longer than user specified column length ({column_length}). "
```

Comment:
Systematic plural is preferred, and whitespace should always be trailing, not leading
```suggestion
                + f"are longer than user specified column length ({column_length}). "
```